### DISCUSSION TITLE
replace suffix when versioning

### DISCUSSION THREAD

**Discussion by mikethecalamity** [renovatebot/renovate#29684] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

In our system, we version our charts based on the version product we're using and append a suffix to handle changes to the charts, when there is no change to the product. For example, if we has a chart for Artemis, when an upgrade of Artemis itself occurs our version would be `2.35.0-0`, but then when we make a change to the chart, that version would become `2.35.0-1`. I am trying to figure out the best way to handle this with Renovate. The suffix should always reset to 0 when Renovate is involved. It seems like `extractVersion` may be what I need, but I don't quite understand the documentation. Some examples below.

**Config**
```json
        {
            "customType": "regex",
            "fileMatch": ["Chart.yaml"],
            "matchStrings": ["datasource=(?<datasource>.*?) depName=(?<depName>.*?)( repository=(?<regisryUrl>.*?))?\\sversion: (?<currentValue>.*)\\s"],
            //versioningTemplate or extractVersionTemplate or currentValueTemplate???
        },
```

***Chart (before)***
```yaml
# renovate: datasource=github-releases depName=apache/activemq-artemis
version: 2.34.0-5
```

***Chart (after)***
```yaml
# renovate: datasource=github-releases depName=apache/activemq-artemis
version: 2.35.0-0
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Do you mean that you _don't_ want Renovate to propose updates unless it ends in `-0`?

**Reply by mikethecalamity** [reply]
No I mean I want Renovate to only check the semver at the front to compare against the latest versions, but then replace the -# suffix with -0. So basically for my Artemis example, Renovate will find `2.34.0-5` in my file, but it should only use `2.34.0` to check against the latest, which is `2.35.0`. Then when it replaces the version, it should replace the whole `2.34.0-5` with `2.35.0-0`.

**Reply by rarkins** [reply]
I'm not sure there's a way to do this.

If you want Renovate to do this then it means we need to treat the entire string as the currentValue/currentVersion. You'll need a versioning like "loose" so that Renovate doesn't treat the -5 and -0 as unstable indicators.

But then your datasource returns e.g. 2.35.0 and not 2.35.0-0. It would work if we found a way to transform 2.35.0 to 2.35.0-0 but I'm not sure we have that.

Potentially you could find a way to hack this using autoReplaceStringTemplate and make the -0 bit static but I don't have time for the experimentation that would take so can't be sure.

**Reply by mikethecalamity** [reply]
@rarkins Have you ever considered adding something like  `newValueTemplate`? Then one could do something like:

```js
newValueTemplate: "{{newValue}}-0"
```
